lib/server.js

```diff
@@ -0,0 +1,198 @@
+var Docker = require('dockerode-promise');
+var debug = require('debug')('docker-exec-websocket-server:lib:server');
+var through = require('through');
+var fs = require('fs');
+var ws = require('ws');
+var slugid = require('slugid');
+var _ = require('lodash');
+var path = require('path');
+var assert = require('assert');
+var url = require('url');
+var msgcode = require('./messagecodes.js');
+
+var dockerSocket = '/var/run/docker.sock';
+var stats = fs.statSync(dockerSocket);
+if (!stats.isSocket()) {
+  throw new Error('Are you sure the docker is running?');
+}
+var docker = new Docker({socketPath: dockerSocket});
+
+// Makes a new buffer with code number (ideally 0x00-0xFF)
+// appended to the start of the given buffer
+function muxBuffer (code, buffer) {
+  return Buffer.concat([new Buffer([code]), buffer]);
+}
+
+/** Makes an instance of docker exec running command, then links the provided
+    websocket to the stream provided by docker exec
+
+    @param tty: boolean representing whether or not terminal emulation is active
+  */
+function execWithSocket (container, socket, command, tty, dbgWrite) {
```

Comment:
Actually turn all of this into a class called something like `class ExecSession {`...
Put validation of configuration in the `constructor` and the `container.exec` and `exec.start` into an `execute` method (async method).

Put message handler, etc into methods too... maybe even some of the pipe work...